### DISCUSSION TITLE
Error newlibc ESP32

### DISCUSSION THREAD

**Discussion by oscarbaselga** [zephyrproject-rtos/zephyr#45269] [Q&A]
Hi all, I am trying to build a Zephyr project for ESP32 which uses libc functions. First option has been set `CONFIG_NEWLIB_LIBC=y`, however, the compilation fails. Maybe newlib libc provided by the Zephyr SDK is not supported in ESP32. I saw that other option is to build the project linking an external provided C library through `CONFIG_EXTERNAL_LIBC=y`. I thought to provide the ESP-IDF newlib libc implementation, but the compilation still fails (I do not know if I am doing it right, I have only added this last flag).

My board is the ESP32 DevKitC, I have the Zephyr SDK v0.13.1, Zephyr v2.7.0 and ESP-IDF v4.3.0

My project is based on hello_world sample. I have only modified the `prj.conf` with previous configuration flags and `main.c` with one call to the `malloc()` function. This same project works for `frdm_k64f` board, so the problem aims to be caused by ESP32.

Thanks ;)

**Comment by stephanosio** [comment]
@sylvioalves
